### ISSUE TITLE
Invalid HTML in header

### ISSUE THREAD

**Issue by ChristianAF** [woocommerce/pinterest-for-woocommerce#455]
### Describe the bug:

> Hi, I have recently installed this plugin to sync products to Pinterest. Everything seemed to have gone ok until I checked our sitechecker software this morning (sitechecker.pro).
> It is now reporting a problem on every page of "Noscript in head contains invalid HTML elements". Apparently this can affect organic search and traffic. I've added a couple of links to the issue below:
> https://sitechecker.pro/site-audit-issues/noscript-head-contains-invalid-html-elements/
> https://sitebulb.com/hints/indexability/head-contains-invalid-html-elements/
> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/noscript
> 
> From what I have read the easiest option is to move the tag out of the head and into the body. Is there anyway to do this within the plugin?

WooCommerce.com Request Number: 252839

**Comment by ChristianAF** [comment]
The tracking tag that is printed by the plugin is based on the base code as given in [Pinterest's documentation.](https://help.pinterest.com/en/business/article/install-the-base-code) 

That base code indeed has an `<img>`  tag in a `<noscript>`  and is meant to be placed in the `<head>`

We probably need to discuss and prioritize any changes in that part.

**Comment by findingsimple** [comment]
@adarshakb getting your eyes on this one for input

**Comment by alexiglesias31** [comment]
@jconroy @adarshakb is there any feedback related to this issue?

**Comment by findingsimple** [comment]
Base code example for reference

```
<head>
 <!-- Pinterest Pixel Base Code -->
 <script type="text/javascript">
   !function(e){if(!window.pintrk){window.pintrk=function(){window.pintrk.queue.push(
     Array.prototype.slice.call(arguments))};var
     n=window.pintrk;n.queue=[],n.version="3.0";var
     t=document.createElement("script");t.async=!0,t.src=e;var
     r=document.getElementsByTagName("script")[0];r.parentNode.insertBefore(t,r)}}("https://s.pinimg.com/ct/core.js");
   pintrk('load', 'YOUR_TAG_ID');
   pintrk('page');
 </script>
 <noscript>
   <img height="1" width="1" style="display:none;" alt="" src="https://ct.pinterest.com/v3/?tid=YOUR_TAG_ID&event=init&noscript=1" />
 </noscript>
 <!-- End Pinterest Pixel Base Code -->
</head>
```

[From here](https://sitechecker.pro/site-audit-issues/noscript-head-contains-invalid-html-elements/)

> Whilst it can be used in both the `<head>` and the `<body>`, when used inside the `<head>`, it must contain only `<link>`,`<style>`, and `<meta>` elements.

@adarshakb can you check internally if the example is correct and if there is a recommendation? e.g. can we remove the noscript, move it elsewhere etc.

**Comment by adarshakb** [comment]
Internal team has a ticket to track this item. Currently no known workaround from my side.

**Comment by adarshakb** [comment]
Internal team said we can move this to the beginning of the Body (first element) to avoid this issue (changing the <noscript> tag part from head into body)

They will update external installation guide. Thanks for bringing this to attention. @jconroy

**Comment by adarshakb** [comment]
Clarifying my comment

leave the script in head but move the noscript to body only


      <head>
       
       <script type="text/javascript">
         !function(e){if(!window.pintrk){window.pintrk=function(){window.pintrk.queue.push(
           Array.prototype.slice.call(arguments))};var
           n=window.pintrk;n.queue=[],n.version="3.0";var
           t=document.createElement("script");t.async=!0,t.src=e;var
           r=document.getElementsByTagName("script")[0];r.parentNode.insertBefore(t,r)}}("https://s.pinimg.com/ct/core.js");
         pintrk('load', 'YOUR_TAG_ID');
         pintrk('page');
       </script>
       
      </head>
      
      <body>
       
       <noscript>
         <img height="1" width="1" style="display:none;" alt="" src="https://ct.pinterest.com/v3/?tid=YOUR_TAG_ID&event=init&noscript=1" />
       </noscript>
       
      </body>

**Comment by dieherbar** [comment]
Working as expected
![Pinterest-After 455 FixApplied](https://user-images.githubusercontent.com/37819277/175081368-4306f956-eaca-4114-9594-09a55ef688fc.png)
![Pinterest-before 455 FixApplied](https://user-images.githubusercontent.com/37819277/175081378-333eb20f-f00d-4a74-9dde-a6aa6b23908b.png)

**Comment by JessixBlade** [comment]
I had two issues related to this today:
zn-5334553
hc-35738513

**Comment by findingsimple** [comment]
@JessixBlade  The fix above was included in https://github.com/woocommerce/pinterest-for-woocommerce/releases/tag/1.2.0 please get the customer to update

**Comment by kuropixel** [comment]
Had one earlier this week. 35918639-hc.

**Comment by findingsimple** [comment]
@kuropixel the fix was released yesterday

I don't think we need to keep capturing instances unless its confirmed the issue still exists after updating

**Comment by kuropixel** [comment]
Thanks @jconroy! Someone had requested that I add it here, apologies!

**Comment by darcie** [comment]
I can still see this after updating to 1.2 on two sites. Example: 
<img width="1083" alt="Screenshot on 2022-06-29 at 19-45-10" src="https://user-images.githubusercontent.com/318285/176581819-a8cd2ef3-fc55-4f92-8036-9fef2e168372.png">

**Comment by findingsimple** [comment]
@darcie that's correct - the noscript was moved from head to within the body

https://github.com/woocommerce/pinterest-for-woocommerce/pull/486

**Comment by darcie** [comment]
@jconroy I get that it was moved, but the `\n\n\n` is still visible on the site. 
I see now that's fixed in #489 and that's not released yet.

**Comment by jconroy** [comment]
Ahhh that _looks_ like a (new) bug that was missed in the review I suspect - can you create a new issue. Is it actually breaking anything?

**Comment by findingsimple** [comment]
> I see now that's fixed in https://github.com/woocommerce/pinterest-for-woocommerce/pull/489 and that's not released yet.

👍

**Comment by darcie** [comment]
I'm pretty sure it's just an appearance issue, but I believe all the recent reports in this are for `\n\n\n` rather than the original issue.

I used the Test Events feature on Pinterest's side and the tag is still tracking. Tag Helper says it's still working. 
Applied the change from #489 and it looks right!

**Comment by sajmes** [comment]
35955356-hc

This was on a site with 1.2.0

**Comment by JPry** [comment]
@sajmes Please see #489, which hasn't been released yet. The original issue described in here has been resolved as of `1.2.0`.
